File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        maxi=nums[0]
        max_so_far=nums[0]
        for i in range(1,len(nums)):
            max_so_far = max(nums[i], max_so_far + nums[i])
            maxi=max(max_so_far,maxi)
        return maxi

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum=0,maxi=Integer.MIN_VALUE;
        // int startInd=-1,endInd=-1,tempStart=-1;

        //Kadane's algorithm
        //with printing
        for(int i=0;i<nums.length;i++){
            // if(sum==0){
            //     tempStart=i;
            // }
            sum+=nums[i];
            if(sum>maxi){
                maxi=sum;
                // startInd=tempStart;
                // endInd=i;
            }
            if(sum<0){
                sum=0;
            }
        }

        // for(int i=startInd;i<=endInd;i++){
        //      System.out.print(nums[i]+" ");
        // }
        return maxi;
    }
}